lib/diagnostic.rb

```diff
@@ -24,7 +24,14 @@
 # be writable.
 
 ##
-# your response here
+Class Person(name, age, location)
+  attr_accessor :name
+  attr_reader :age
+  attr_writer :location
+    @name = name
+    @age = age
+    @location = location
+end
```

Comment:
Very close! The parameters with which you initialize instances of a class belong in the `initialize` method.